supervision/utils/internal.py

```diff
@@ -1,5 +1,37 @@
 import functools
 import warnings
+from typing import Callable
+
+
+def deprecated_parameter(
+    old_param: str, new_param: str, map_func: Callable = lambda x: x
```

Comment:
I like the idea. When would be the example use case when I'd use the non-default `map_func` value?

Reply:
Will this work with MyPy?

Reply:
Hi!
> I like the idea. When would be the example use case when I'd use the non-default map_func value?

The function is intended to map between types, for example, we use it to map from `Position` to `Iterable[Position]` [here](Iterable[Position]). It could also be used in more complex parameter changes like bounding_box formats e.g. [x1, y1, x2, y2] -> [x_center, y_center, w, h].

> Will this work with MyPy?

Good question. To further analyze, please consider this MyPy [playground sample](https://mypy-play.net/?mypy=latest&python=3.12&gist=b6cc4856cd0a2a0a2270c0e69a2bb80b).
MyPy finds no issues with the code. However, It is still possible to use the `map_function` to cast the old parameter to a inconsistent type: [example](https://mypy-play.net/?mypy=latest&python=3.12&gist=d486f2a2f2420c82f84df9e85ea889cd). It also does not find any issues in the second case.

Reply:
Understood! Awesome!